subsys/net/l2/ethernet/lldp/lldp.c

```diff
@@ -120,6 +120,16 @@ static int lldp_send(struct ethernet_lldp *lldp)
 		}
 	}
 
+#if defined(CONFIG_NET_LLDP_END_LLDPDU_TLV_ENABLED)
+	u16_t tlv_end = htons(NET_LLDP_END_LLDPDU_VALUE);
```

Comment:
never declare a variable in the middle of a code block. Adding { before and } after line 130 could be a solution.